### PR TITLE
subcommand api

### PR DESCRIPTION
**Pull request by shakfu**
Introduces a new subcommand api:

```
shedskin analyze <target>
shedskin translate <target>
shedskin build <target>
shedskin run <target>
shedskin test
```
Note that `shedskin translate` is the exactly the same as the prior api and shedskin `build, run and test` all require cmake

### CODE REVIEW

**Comment by srepmub** [comment]
oof, that was quite a bit of work.. :) thanks a lot! I won't look at everything in detail, but as long as all tests/examples still work (and shedskin translate provides the old behavior), I trust your judgment. I will let you know when I run into something while using shedskin.

**Comment by shakfu** [comment]
@srepmub 

Thanks for the confidence!

Incidentally, I think we need to revisit the requirement set by shedskin.graph.parse_module which makes it a requirement for the input name to be a module in the cwd.

shedskin.set_name and a lot of code basically is jumping through hoops to get around this,  and while the current implementation worlks, it is more of a hackish work-around than a solid solution.

Basically the input value to shedskin should be a full path to a module rather than a name. This makes it easier to handle over the long run.
